docs/index.rst

```diff
@@ -106,23 +106,79 @@ You can use pip and make to quickly set up a development environment::
+* If PR is applicable for many branches (e.g. master and one of '6.X.z' branches), specify that information in PR message
+* Unit tests
+    * Unit tests are compulsory
+    * Unit tests should cover all available actions, for the entity. For eg: Repository Sets, have enable, disable, list_available there should be unit tests exercising these actions.
+* When in doubt, ask on IRC. Join #robottelo on `freenode`_
+
+**Important to Note :**
+
+* Define Foreman Version labels in Nailgun
+    if possible, the contributor should set the right version.
+
+* All PRs should be raised along with Unit tests
+    The unit tests should be added while adding a new entity or modifying the existing entity or modifying and adding to the core of Nailgun.
+
+* Test results from upstream devel or from upstream nightly
+    The API call results are required to make the review process more firm. The PR author can provide results from any library that uses the contributed code by running the changes on upstream nightly or from his/her devel box. The interactive python shell output would be acceptable
+
+Nailgun Review Process
+______________________
+
+* Travis CI is run, and any issues are resolved by contributor.
+* If deemed necessary by contributors/reviewers, an automation run is triggered.
+* At least two ACKs are required to merge a pull request.
+* At least one ACK must be from a Tier 2 reviewer.
+* If a PR requires changes to the CI environment, the “CI Owner” must also provide an ACK.
+* Pull request is merged by the person who does the final required ACK.
+
+**Reviewers & Responsibilities :**
+
+* Both Tiers
+    * Consistently check your projects for new pull requests.
+    * Check code for consistency with project guidelines.
+    * Pin code dependencies (external libraries), against the version it was tested.
+    * Determine if CI and/or test infrastructure changes are required.
+    * Provide helpful feedback.
+    * Follow-up with any pending feedback, to ensure the PR is resolved quickly.
+* Tier 1 Reviewer
+    * Check the scenarios are valid for the feature or components
+    * Suggestions on the feature that can be covered with minimal code additions/changes.
+* Tier 2 Reviewer
+    * Check for logical errors.
+    * Guide the contributor on how to fix mistakes and any other improvements.
+    * Ideally if not done by contributor, identify code that may impact third-party projects (e.g Nailgun -> Robottelo , FAM), file issues if PR causes breakages in relevant projects
+
+
+Nailgun Release Process
+_______________________
+
+
+Projects that require nailgun, would often rely on the released Nailgun from Pypi.
+We intended to make the release process more formal and standard to deliver timely and stable code base to consumer projects.
+
+* Nailgun Releases should be performed against stable branches.
+* No historical release support.
+* Nailgun will follow request based minor releases.
+
 
-* When in doubt, ask on IRC.
```

Comment:
I would update that list, because it is very-very old:
```
.. _Foreman bug #4638: http://projects.theforeman.org/issues/4638
.. _freenode: https://freenode.net/
.. _good commit messages: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
.. _Rebasing: http://www.git-scm.com/book/en/v2/Git-Branching-Rebasing
.. _Requests: http://docs.python-requests.org/en/latest/
.. _Robottelo: http://robottelo.readthedocs.io/en/latest/
.. _Robottelo source code: https://github.com/SatelliteQE/robottelo
.. _this blog post: http://www.ichimonji10.name/blog/4/
```
to at least:
```
.. _good commit messages: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
.. _Rebasing: http://www.git-scm.com/book/en/v2/Git-Branching-Rebasing
.. _Requests: http://docs.python-requests.org/en/latest/
.. _Robottelo: http://robottelo.readthedocs.io/en/latest/
.. _Robottelo source code: https://github.com/SatelliteQE/robottelo
```

Reply:
removed the links that are not needed